fbw-a380x/src/systems/instruments/src/MFD/pages/common/TopTabNavigator.tsx

```diff
@@ -0,0 +1,254 @@
+          />
+          {!this.props.isSelected && (
+            <line
+              x1="0"
+              y1={this.props.height - 1}
+              x2={this.triangleWidth}
+              y2={this.props.height - 1}
+              style="stroke: lightgrey; stroke-width:2"
+            />
+          )}
+        </svg>
+        <span
+          ref={this.textRef}
+          class={`mfd-top-tab-navigator-bar-element-label${this.props.isSelected ? ' active' : ''}`}
+          // eslint-disable-next-line max-len
+          style={`font-size: ${Math.floor(this.props.height * 0.55)}px;`}
+        >
+          {this.props.title}
+        </span>
+        <svg height={this.props.height} width={this.triangleWidth}>
+          <polygon
+            points={`0,0 ${this.triangleWidth},${this.props.height} 0,${this.props.height}`}
+            style={`fill:${this.props.isSelected ? '#000000' : '#3c3c3c'};`}
+          />
+          <line
+            x1="0"
+            y1="0"
+            x2={this.triangleWidth}
+            y2={this.props.height}
+            style="stroke: lightgrey; stroke-width:2"
+          />
+          {!this.props.isSelected && (
+            <line
+              x1="0"
+              y1={this.props.height - 1}
+              x2={this.triangleWidth}
+              y2={this.props.height - 1}
+              style="stroke: lightgrey; stroke-width:2"
+            />
+          )}
+        </svg>
+      </div>
+    );
+  }
+}
+
+interface TopTabNavigatorPageProps {
+  isVisible?: Subscribable<boolean>;
+  containerStyle?: string;
+}
+
+export class TopTabNavigatorPage extends DisplayComponent<TopTabNavigatorPageProps> {
+  private topDivRef = FSComponent.createRef<HTMLDivElement>();
+
+  private isVisible = false;
+
+  public getVisibility(): boolean {
+    return this.isVisible;
+  }
+
+  public setVisibility(value: boolean) {
+    this.isVisible = value;
+
+    if (this.topDivRef.instance !== undefined) {
+      this.topDivRef.instance.style.display = value ? 'flex' : 'none';
+    }
+  }
+
+  render(): VNode {
+    return (
+      <div ref={this.topDivRef} class="mfd-top-tab-navigator-tab-content" style={this.props.containerStyle}>
+        {this.props.children}
+      </div>
+    );
+  }
+}
+
+interface TopTabNavigatorProps {
+  pageTitles: Subscribable<string[]>;
+  selectedPageIndex: Subject<number>;
+  selectedTabTextColor?: string;
+  tabBarHeight?: number; // in pixels
+  tabBarSlantedEdgeAngle?: number; // in degrees, vertical line equals 0°
+  additionalRightSpace?: number; // in pixels
+  activeFlightPhase?: Subscribable<FmgcFlightPhase>; // special handling for PERF pages, mark tab from active flight phase with green text color
+  pageChangeCallback?: (index: number) => void;
+}
+
+/*
+ * Container for multiple tabs, with tab navigation at the top. Pages are instantiated as children of the component, as TopTabElement
+ */
+export class TopTabNavigator extends DisplayComponent<TopTabNavigatorProps> {
```

Comment:
Are you referring to TopTabNavigatorProps? It only accepts props relevant to the styling of the tabs at the top, but not relevant to the styling of the children (i.e. the page content)

Reply:
Yeah, I meant the tabs at the top, but I didn't realize that the children were the actual pages 😄. I've not tried this myself, but I was thinking of a structure like this. The goal is that the generic component for a tabbed layout should not know about aircraft things such as the flight phase at all, it's a UI component. If you need specific styling for the tab of the active flight phase, you could pass in a color from outside.

```jsx
<TopTabLayout>
  <TopTabNavigator>
    <TopTabElement pageTitle="CLB" ... />
    <TopTabElement pageTitle="CRZ" ... />
  </TopTabNavigator>
  <TopTabContent>
    {/* pages */}
  </TopTabContent>
</TopTabLayout>
```

Reply:
I tried it, it makes it very awkward since you would have to duplicate a lot of props for each ob the sub-components, which blows up the code in the FMS page tsx files. 
Popular libs such as react-navigation also use styling props at the top level: https://reactnavigation.org/docs/material-top-tab-navigator

What I can do though, is generalize activeFlightPhase into something which accepts an array of text colors for the tabs

Reply:
> What I can do though, is generalize activeFlightPhase into something which accepts an array of text colors for the tabs

Yep, I like this. I just don't want a generic UI component to know about something like a flight phase.

Reply:
I now introduced "highlightedTab", which marks the selected tab index with green text color. No need to be generic, as far as I could see that's the only instance where this happens.

Reply:
Excellent 👍